src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -99,23 +149,155 @@ const SKUpdate& SKNMEA2000Parser::parse128267(const SKSourceInput& input, const
 
     update->setEnvironmentDepthBelowTransducer(depthBelowTransducer);
 
-    if (!N2kIsNA(offset)) {
-      // When offset is negative, it's the distance between transducer and keel
-      if (offset < 0) {
-        update->setEnvironmentDepthTransducerToKeel(offset * -1);
-        update->setEnvironmentDepthBelowKeel(depthBelowTransducer + offset);
-      }
-      else if (offset > 0) {
-        update->setEnvironmentDepthSurfaceToTransducer(offset);
-        update->setEnvironmentDepthBelowSurface(depthBelowTransducer + offset);
+    if (N2kIsNA(offset)) {
+      // When no offset then offset should be Zero
+      offset = 0;
+    }
+    // When offset is negative, it's the distance between transducer and keel
+    if (offset < 0) {
```

Comment:
Am I missing something? Or is this is not going to work when the offset is 0. I am surprised the tests did not catch this, we should have a test for this.